architecture/jvmfrc/src/main/kotlin/com/lynbrookrobotics/kapuchin/hardware/imu/ADIS16448.kt

```diff
@@ -0,0 +1,128 @@
+package com.lynbrookrobotics.kapuchin.hardware.imu
+
+import com.lynbrookrobotics.kapuchin.control.data.UomVector
+import com.lynbrookrobotics.kapuchin.hardware.SPI
+import info.kunalsheth.units.generated.Angle
+import info.kunalsheth.units.generated.AngularVelocity
+import info.kunalsheth.units.generated.Degree
+import info.kunalsheth.units.generated.DegreePerSecond
+import java.nio.ByteBuffer
+
+
+/**
+ * An interface for communicating with the ADIS16448 IMU.
+ *
+ * @author Nikash Walia
+ *
+ * @param spi the SPI object being used to communicate with the gyro
+ * @param updatePeriod updatePeriod of the clock
+ */
+class ADIS16448(private val spi: SPI, updatePeriod: Double = 1.0 / 3000000): DigitalGyro(updatePeriod) {
```

Comment:
On our robots, the ADIS16448 will always be hooked up to the RoboRIO MXP.
The ADIS16448 is in `jvmfrc`, which is already a platform dependent module.
Thus, I think the SPI abstraction is unnecessary. Instead, make the ADIS16448 hold an instance of the [WPILIB Serial class](http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/SerialPort.html).